woocommerce/woocommerce
# Adds geo and install plugin control to extensions featured page

On the featured tab of the Extensions page we show extensions.

- This PR hides extensions that are activated on the site.
- This PR also adds the concept of geo blacklist and whitelists.

The featured.json version bumped to 2 because adding more than three extensions per section will show all of them in previous versions.  We need to add more than three extensions for when extensions are not shown due to the above constraints.

In order to test:

- Go to featured tab of the extensions page
- Observe that any extensions that you have activated on that site are not shown

To test the geo gating you will need to download the featured-v2.json file and add:

"geowhitelist": [ "GB" ] or "geoblacklist": [ "US" ] to one or more of the extensions.  Do not add both to one since that wouldn't make any sense. Once you have the modified file put it somewhere web accessible and change the url on line 27 of includes/admin/class-wc-admin-addons.php.

A whitelist is nothing but the countries in the list.
A blacklist is everything but the countries in the list.
